sdks/python/apache_beam/runners/interactive/background_caching_job.py

```diff
@@ -103,6 +102,7 @@ def cancel(self):
     if not PipelineState.is_terminal(self._pipeline_result.state):
       try:
         self._pipeline_result.cancel()
+        self._pipeline_result.wait_until_finish()
```

Comment:
I'm sure you've considered this, but just to be sure, have you given thought to other code assuming that this call is asynchronous?
Also, might this pipeline run in an environment where a cancelation may take a long time? I guess it shouldnt be a problem generally?

Reply:
Yeah, in a later PR I added a lock about the pipeline result because of this comment. And it generally shouldn't be a problem if the cancellation takes a long time.